ReactiveTask/Task.swift

```diff
@@ -261,32 +385,76 @@ public func launchTask(taskDescription: TaskDescription, standardOutput: SinkOf<
-				let stdoutProducer = aggregateDataReadFromPipe(stdoutPipe, standardOutput)
+			|> flatMap(.Concat) { stdoutPipe -> SignalProducer<TaskEvent<NSData>, ReactiveTaskError> in
+				let stdoutProducer = aggregateDataReadFromPipe(stdoutPipe)
 
 				return SignalProducer(result: Pipe.create(queue))
-					|> flatMap(.Merge) { stderrPipe -> SignalProducer<NSData, ReactiveTaskError> in
-						let stderrProducer = aggregateDataReadFromPipe(stderrPipe, standardError)
+					|> flatMap(.Merge) { stderrPipe -> SignalProducer<TaskEvent<NSData>, ReactiveTaskError> in
+						let stderrProducer = aggregateDataReadFromPipe(stderrPipe)
+
+						return SignalProducer { observer, disposable in
+							let (stdoutAggregated, stdoutAggregatedSink) = SignalProducer<NSData, ReactiveTaskError>.buffer(1)
+							let (stderrAggregated, stderrAggregatedSink) = SignalProducer<NSData, ReactiveTaskError>.buffer(1)
+
+							stdoutProducer.startWithSignal { signal, signalDisposable in
+								disposable += signalDisposable
+
+								signal.observe(next: { readData in
+									switch readData {
+									case let .Chunk(data):
+										sendNext(observer, .StandardOutput(data))
+
+									case let .Aggregated(data):
+										sendNext(stdoutAggregatedSink, data)
+									}
+								}, error: { error in
+									sendError(observer, error)
+									sendError(stdoutAggregatedSink, error)
+								}, completed: {
+									sendCompleted(stdoutAggregatedSink)
+								}, interrupted: {
+									sendInterrupted(stdoutAggregatedSink)
+								})
+							}
 
-						let terminationStatusProducer = SignalProducer<Int32, NoError> { observer, disposable in
-							task.terminationHandler = { task in
-								sendNext(observer, task.terminationStatus)
-								sendCompleted(observer)
+							stderrProducer.startWithSignal { signal, signalDisposable in
+								disposable += signalDisposable
+
+								signal.observe(next: { readData in
+									switch readData {
+									case let .Chunk(data):
+										sendNext(observer, .StandardError(data))
+
+									case let .Aggregated(data):
+										sendNext(stderrAggregatedSink, data)
+									}
+								}, error: { error in
+									sendError(observer, error)
+									sendError(stderrAggregatedSink, error)
+								}, completed: {
+									sendCompleted(stderrAggregatedSink)
+								}, interrupted: {
+									sendInterrupted(stderrAggregatedSink)
+								})
 							}
 
 							task.standardOutput = stdoutPipe.writeHandle
 							task.standardError = stderrPipe.writeHandle
 
-							if disposable.disposed {
-								stdoutPipe.closePipe()
-								stderrPipe.closePipe()
-
-								// Clean up the stdin pipe in a roundabout way.
-								stdinProducer.startWithSignal { signal, signalDisposable in
-									signalDisposable.dispose()
+							task.terminationHandler = { task in
+								let terminationStatus = task.terminationStatus
+								if terminationStatus == EXIT_SUCCESS {
```

Comment:
When `terminationStatus != EXIT_SUCCESS`, signal from `stderrProducer` does not wait signal from `stdoutProducer`. Signal from other hand does not be needed, but should be waited.
I guess this cause segmentation fault of https://github.com/Carthage/Carthage/issues/497